File: scripts/generate-demo-qr.py
#!/usr/bin/env python3
"""生成演示联系二维码占位图 public/placeholders/demo-contact-qr.png"""
from pathlib import Path

try:
    import qrcode
except ImportError:
    raise SystemExit("请先安装: pip install qrcode[pil]")

ROOT = Path(__file__).resolve().parents[1]
OUT = ROOT / "public" / "placeholders" / "demo-contact-qr.png"

def main() -> None:
    OUT.parent.mkdir(parents=True, exist_ok=True)
    img = qrcode.make("demo-placeholder-resume-comparison", box_size=8, border=2)
    img.save(OUT)
    print(f"Saved: {OUT}")

if __name__ == "__main__":
    main()

File: scripts/generate-demo-qr.mjs
#!/usr/bin/env node
/** 生成演示用占位二维码 PNG（扫描内容为 demo-placeholder，无实际跳转） */
import path from "path";
import { fileURLToPath } from "url";
import QRCode from "qrcode";

const __dirname = path.dirname(fileURLToPath(import.meta.url));
const out = path.join(__dirname, "..", "public", "placeholders", "demo-contact-qr.png");

await QRCode.toFile(out, "demo-placeholder", {
  width: 320,
  margin: 2,
  color: { dark: "#1a1a1a", light: "#ffffff" },
});

console.log(`Wrote ${path.relative(path.join(__dirname, ".."), out)}`);
